test/test_functional.py

```diff
@@ -0,0 +1,169 @@
+
+    def test_istft_is_inverse_of_stft5(self):
+        # hamming_window, not centered, not normalized, not onesided
+        # window same size as n_fft
+        kwargs5 = {
+            'n_fft': 3,
+            'hop_length': 2,
+            'win_length': 3,
+            'window': torch.hamming_window(3),
+            'center': False,
+            'pad_mode': 'reflect',
+            'normalized': False,
+            'onesided': False,
+        }
+
+        self._test_istft_is_inverse_of_stft(kwargs5)
+
+    def test_istft_of_ones(self):
+        # stft = torch.stft(torch.ones(4), 4)
+        stft = torch.tensor([
+            [[4., 0.], [4., 0.], [4., 0.], [4., 0.], [4., 0.]],
+            [[0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.]],
+            [[0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.]]
+        ])
+
+        estimate = torchaudio.functional.istft(stft, n_fft=4, length=4)
+        self._compare_estimate(torch.ones(4), estimate)
+
+    def test_istft_of_zeros(self):
+        # stft = torch.stft(torch.zeros(4), 4)
+        stft = torch.zeros((3, 5, 2))
+
+        estimate = torchaudio.functional.istft(stft, n_fft=4, length=4)
+        self._compare_estimate(torch.zeros(4), estimate)
+
+    def test_istft_requires_overlap_windows(self):
+        # the window is size 1 but it hops 20 so there is a gap which throw an error
+        stft = torch.rand((3, 5, 2))
+        self.assertRaises(AssertionError, torchaudio.functional.istft, stft, n_fft=4,
+                          hop_length=20, win_length=1, window=torch.ones(1))
+
+    def test_istft_requires_nola(self):
+        stft = torch.zeros((3, 5, 2))
+        kwargs_ok = {
+            'n_fft': 4,
+            'win_length': 4,
+            'window': torch.ones(4),
+        }
+
+        kwargs_not_ok = {
+            'n_fft': 4,
+            'win_length': 4,
+            'window': torch.zeros(4),
+        }
+
+        # A window of ones meets NOLA but a window of zeros does not. This should
+        # throw an error.
+        torchaudio.functional.istft(stft, **kwargs_ok)
+        self.assertRaises(AssertionError, torchaudio.functional.istft, stft, **kwargs_not_ok)
+
+    def test_istft_of_sine(self):
+        # stft of 123*sin(2*pi/5*x) with the hop length and window size equaling the period of L = 5
+        x = torch.arange(10, dtype=torch.get_default_dtype())
+        L = 5
+        amplitude = 123
+        sound = amplitude * torch.sin(2 * math.pi / L * x)
+        # stft = torch.stft(sound, L, hop_length=L, win_length=L,
+        #                   window=torch.ones(L), center=False, normalized=False)
+        stft = torch.zeros((3, 2, 2))
+        stft[1, :, 1] = -(amplitude * L) / 2.0
+
+        estimate = torchaudio.functional.istft(stft, L, hop_length=L, win_length=L,
+                                               window=torch.ones(L), center=False, normalized=False)
+        self._compare_estimate(sound, estimate, atol=1e-4)
```

Comment:
Please use a consistent tolerance value by reducing this to `1e-6` as done elsewhere in this code.

Reply:
An issue here is that the stft is actually 1e-5 around everything that is not the main frequency of the signal (not actually zero). So I'm guessing the fft of a pure sine wave still has very tiny bit of noise. The other tests use 1e-6 because we feed the output of stft directly into istft.
Example:
```
# amplitude=234, L=5, n=2
tensor([[[-3.0518e-05,  0.0000e+00],
         [ 8.6690e-05,  0.0000e+00]],

        [[ 4.1749e-05, -5.8500e+02],
         [ 7.2119e-05, -5.8500e+02]],

        [[-2.6490e-05,  3.1527e-05],
         [-1.3180e-05,  1.5360e-04]]])
```

Reply:
Actually I think It's the scaling of the signal. 0.001 error for a signal of ~100 amplitude in this test instead of 0.000001 error for a signal of amplitude ~1 in the other tests.

Reply:
In this case, let's specify the relative tolerance of 1e-5 instead of the absolute tolerance as it conveys what you are saying here more explicitly.